perfkitbenchmarker/linux_benchmarks/memcached_ycsb_benchmark.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs YCSB against memcached.
+
+This benchmark runs two workloads against memcached using YCSB (the Yahoo! Cloud
+Serving Benchmark).
+memcached is described in perfkitbenchmarker.linux_packages.memcached_server
+YCSB and workloads described in perfkitbenchmarker.linux_packages.ycsb.
+"""
+
+import functools
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.linux_packages import memcached_server
+from perfkitbenchmarker.linux_packages import ycsb
+
+FLAGS = flags.FLAGS
+
+
+BENCHMARK_NAME = 'memcached_ycsb'
+BENCHMARK_CONFIG = """
+memcached_ycsb:
+  description: >
+    Run YCSB against an memcached
+    installation. Specify the number of YCSB client VMs with
+    --ycsb_client_vms and the number of YCSB server VMS with
+    --ycsb_server_vms.
+  vm_groups:
+    servers:
+      vm_spec: *default_single_core
+    clients:
+      vm_spec: *default_single_core
+"""
+
+
```

Comment:
I'll add that stuff in a followup PR. I have some more code that adds managed backends to this, and I want to provide configuration options that are equivalent across all providers.